nengo_spa/compiler/tests/test_ast.py

```diff
@@ -0,0 +1,436 @@
+from io import BytesIO
+
+import pytest
+
+import nengo_spa as spa
+from nengo_spa.exceptions import SpaTypeError
+from nengo_spa.compiler import tokens
+from nengo_spa.compiler.ast import AstBuilder
+from nengo_spa.compiler.ast_nodes import (
+    Action, ActionSet, ApproxInverse, DotProduct, Effect, Effects, Negative,
+    Module, Product, Scalar, Sink, Sum, Symbol)
+from nengo_spa.compiler.ast_types import (
+    TActionSet, TEffect, TEffects, TScalar, TVocabulary)
+
+
+def bare_tokens(expr):
```

Comment:
Maybe explain why this is used for tests that don't need a full parse tree structure?